eZ/Publish/Core/Repository/Repository.php

```diff
@@ -784,23 +790,7 @@ protected function getPermissionCriterionResolver(): PermissionCriterionResolver
      */
     protected function getCachedPermissionsResolver()
     {
-        if ($this->permissionsHandler === null) {
-            $this->permissionsHandler = new CachedPermissionService(
-                $permissionResolver = new Permission\PermissionResolver(
-                    $this->getRoleDomainMapper(),
-                    $this->limitationService,
-                    $this->persistenceHandler->userHandler(),
-                    new UserReference($this->serviceSettings['user']['anonymousUserID']),
-                    $this->serviceSettings['role']['policyMap']
-                ),
-                new PermissionCriterionResolver(
-                    $permissionResolver,
-                    $this->limitationService
-                )
-            );
-        }
-
-        return $this->permissionsHandler;
+        return $this->permissionResolver;
```

Comment:
Hmm, what we've returned here previously was an instance which implemented both `\eZ\Publish\API\Repository\PermissionResolver` and **`\eZ\Publish\API\Repository\PermissionCriterionResolver`**.
I see that the correct instance is injected, but it causes a type mismatch here.
What I did yesterday was an introduction of a common API extending both of those interfaces so we don't introduce a technical debt.
```php
namespace eZ\Publish\API\Repository;

/**
 * Marker for cached Permission Resolver.
 */
interface PermissionService extends PermissionResolver, PermissionCriterionResolver
{
}
```

```diff
-class CachedPermissionService implements APIPermissionResolver, APIPermissionCriterionResolver
+class CachedPermissionService implements PermissionService
```

Reply:
Proposed in #46